src/main/java/org/monarchinitiative/biodownload/BioDownloaderBuilder.java

```diff
@@ -0,0 +1,224 @@
+        return this;
+    }
+
+    public BioDownloaderBuilder mondo() {
+        return mondoOwl().mondoJson();
+    }
+
+    public BioDownloaderBuilder ectoJson() {
+        resources.add(DownloadableResource.ECTO_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder ectoOwl() {
+        resources.add(DownloadableResource.ECTO_OWL);
+        return this;
+    }
+
+    public BioDownloaderBuilder ecto() {
+        return ectoOwl().ectoJson();
+    }
+
+    public BioDownloaderBuilder maxoJson() {
+        resources.add(DownloadableResource.MAXO_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder maxoOwl() {
+        resources.add(DownloadableResource.MAXO_OWL);
+        return this;
+    }
+
+    public BioDownloaderBuilder maxoObo() {
+        resources.add(DownloadableResource.MAXO_OBO);
+        return this;
+    }
+
+    public BioDownloaderBuilder maxo() {
+        return maxoOwl().maxoObo().maxoJson();
+    }
+
+    public BioDownloaderBuilder hpoJson() {
+        resources.add(DownloadableResource.HP_JSON);
+        return this;
+    }
+
+    public BioDownloaderBuilder hpoObo() {
+        resources.add(DownloadableResource.HP_OBO);
+        return this;
+    }
+
+    public BioDownloaderBuilder hpo() {
+        return hpoObo().hpoJson();
+    }
+
+    public BioDownloaderBuilder hpAnnotation() {
+        resources.add(DownloadableResource.PHENOTYPE_HP);
+        return this;
+    }
+
+    public BioDownloaderBuilder geneInfo() {
+        resources.add(DownloadableResource.GENE_INFO);
+        return this;
+    }
+
+    public BioDownloaderBuilder all() {
+        return go().goGaf().goGafGz().hpo().mondo().ecto().hgnc().maxo().medgene2MIM().proSite().hpAnnotation().geneInfo();
+    }
+
+    /**
+     * Download file from given URL and save it with the given name
+     * @param name Name of file to be downloaded to
+     * @param urlStr URL string of source file
+     * @return a builder instance
+     */
+    public BioDownloaderBuilder custom(String name, String urlStr) {
+        URL url = DownloadableResource.createURL(urlStr);
+        resources.add(new DownloadableResource(name, url));
+        return this;
+    }
+
+    /**
+     * Build Downloader
+     * @return a {@link IBioDownloader} object
+     */
+    public IBioDownloader build() {
+        validate();
+        return new BioDownloaderImpl(resources, destinationPath, overwrite);
+    }
+
+    /**
+     * Validation of the Builder
+     * @throws IllegalStateException If the validation fails we throw an {@link IllegalStateException}
+     */
+    private void validate() throws IllegalStateException {
+        StringBuilder sb = new StringBuilder();
+        if (resources.size() == 0) {
+            sb.append("A name and a URL need to be included. Please pick one of the available options or add your custom name and URL.");
+        }
+        for (DownloadableResource resource : resources) {
```

Comment:
This check is appropriate right now, but no action is taken to remove the (I think) useless null-containing `DownloadableResource`.

Reply:
Actually I see now that the presence of `null` will throw `IllegalStateException`.